aptos-move/framework/aptos-stdlib/sources/cryptography/ed25519.move

```diff
@@ -166,6 +170,43 @@ module aptos_std::ed25519 {
         std::hash::sha3_256(pk_bytes)
     }
 
+    #[test_only]
+    fun generate_keys(): (SecretKey, UnvalidatedPublicKey) {
+        let (sk_bytes, pk_bytes) = generate_keys_internal();
+        let sk = SecretKey {
+            bytes: sk_bytes
+        };
+        let pk = UnvalidatedPublicKey {
+            bytes: pk_bytes
+        };
+        (sk,pk)
+    }
+
+    #[test_only]
+    fun sign_arbitrary_bytes(sk: &SecretKey, msg: vector<u8>): Signature {
+        Signature {
+            bytes: sign_internal(sk.bytes, msg)
+        }
+    }
+
+    #[test_only]
+    fun sign_struct<T:drop>(sk: &SecretKey, data: T): Signature {
+        let encoded = new_signed_message(data);
+        Signature {
+            bytes: sign_internal(sk.bytes, bcs::to_bytes(&encoded))
+        }
+    }
+
+    #[test]
+    fun test_gen_sign_verify_combo() {
+        let (sk, pk) = generate_keys();
+        let msg: vector<u8> = x"0123456789abcdef";
+        let sig1 = sign_arbitrary_bytes(&sk, msg);
+        assert!(signature_verify_strict(&sig1, &pk, msg), std::error::invalid_state(1));
```

Comment:
When writing tests, we usually just pass 0 to the last argument of `assert`.

Some times, we pass 0 for the first `assert` call, 1 for the second, and so on.

But we usually don't bother with using the right `std::error` codes **in tests**. (We do bother in actual non-testing code.)